File: com.systemincloud.examples.tasks.javatask/src/test/java/tasks/generator/GeneratorLong.java
package tasks.generator;

import com.systemincloud.modeler.tasks.javatask.api.JavaTask;
import com.systemincloud.modeler.tasks.javatask.api.OutputPort;
import com.systemincloud.modeler.tasks.javatask.api.annotations.JavaTaskInfo;
import com.systemincloud.modeler.tasks.javatask.api.annotations.OutputPortInfo;
import com.systemincloud.modeler.tasks.javatask.api.data.Control;

@JavaTaskInfo(generator = true)
public class GeneratorLong extends JavaTask {

	@OutputPortInfo(name = "Out", dataType = Control.class)
	public OutputPort out;

	@Override
	public void generate() {
		sleep(3000);
		try { Thread.sleep(2000); } catch (InterruptedException e) { }
		out.putData(new Control());
	}
}

File: com.systemincloud.examples.tasks.pythontask/src/test/py/tasks/generator/GeneratorLong.py
from sicpythontask.PythonTaskInfo import PythonTaskInfo
from sicpythontask.PythonTask import PythonTask
from sicpythontask.OutputPort import OutputPort
from sicpythontask.data.Control import Control
import time

@PythonTaskInfo(generator=True)
class GeneratorLong(PythonTask):

    def __init_ports__(self):
        self.out = OutputPort(name="out")

    def generate(self):
        self.sleep(3000)
        time.sleep(2)
        self.out.put_data(Control());
